src/api/0x/index.ts

```diff
@@ -0,0 +1,136 @@
+import { OrderKind } from '@cowprotocol/contracts'
+import log from 'loglevel'
+import { SupportedChainId } from '../../constants/chains'
+import { Context } from '../../utils/context'
+import { getTokensFromMarket } from '../../utils/market'
+import { toErc20Address } from '../../utils/tokens'
+
+import BaseApi from '../base'
+
+import { Options, PriceQuoteParams } from '../cow/types'
+import ZeroXError, { logPrefix, ZeroXErrorResponse } from './error'
+import { ERC20BridgeSource, MatchaOptions, MatchaPriceQuote } from './types'
+
+// GPV2Settlement
+// https://etherscan.io/address/0x9008d19f58aabd9ed0d60971565aa8510560ab41
+const AFFILIATE_ADDRESS = '0x9008D19f58AAbD9eD0D60971565AA8510560ab41'
+const EXCLUDED_SOURCES: ERC20BridgeSource[] = []
+const API_VERSION = 'v1'
+
+function get0xUrls(): Partial<Record<SupportedChainId, string>> {
+  // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+  // See https://0x.org/docs/api#introduction
+  return {
+    [SupportedChainId.MAINNET]: 'https://api.0x.org/swap',
+  }
+}
+
+function _getMatchaChainId(chainId: SupportedChainId): SupportedChainId | null {
+  switch (chainId) {
+    // Support: Mainnet, Ropsten, Polygon, Binance Smart Chain
+    // See https://0x.org/docs/api#introduction
+    // but we only support mainnet of that list so..
+    case SupportedChainId.MAINNET:
+      return chainId
+
+    default:
+      // Unsupported network
+      return null
+  }
+}
+
+function _optionsToSearchUrl(options: MatchaOptions) {
+  const { excludedSources, affiliateAddress } = options
+
+  const excludedSourceString = excludedSources.length > 0 ? excludedSources.join(',') : ''
+  const optionsMap = Object.entries({ excludedSources: excludedSourceString, affiliateAddress }).map(
+    ([option, value]) => `${option}=${value}`
+  )
+  return optionsMap.join('&')
```

Comment:
could this utility function work here? https://github.com/cowprotocol/cow-sdk/blob/44e38ba32db433778836b2d47a55bca61411bfbf/src/utils/common.ts#L10

Reply:
I see this is concatenated later in:

```js
 `/price?sellToken=${sellToken}&buyToken=${buyToken}&${swapSide}=${amount}&${this.MATCHA_OPTIONS_URL}`
```

and this function should we called just once with every qs parameter. Maybe handling all the parameters as an object?

Reply:
think so, will adapt my code